sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/publishing/InfluxDBPublisher.java

```diff
@@ -146,16 +157,29 @@ private static HttpPost providePOSTRequest(final InfluxDBSettings settings) {
   private static void executeWithVerification(
       final HttpPost postRequest, final HttpClientBuilder builder) throws IOException {
     try (final CloseableHttpResponse response = builder.build().execute(postRequest)) {
-      is2xx(response.getStatusLine().getStatusCode());
+      is2xx(response);
     }
   }
 
-  private static void is2xx(final int code) throws IOException {
+  private static void is2xx(final HttpResponse response) throws IOException {
+    final int code = response.getStatusLine().getStatusCode();
     if (code < 200 || code >= 300) {
-      throw new IOException("Response code: " + code);
+      throw new IOException(
+          "Response code: " + code + ". Reason: " + getErrorMessage(response.getEntity()));
     }
   }
 
+  private static String getErrorMessage(final HttpEntity entity) throws IOException {
+    final Header encodingHeader = entity.getContentEncoding();
```

Comment:
If you want to make it even more secure and resilient, you can check what's the value of `Content-Type` header. For JSON, it must be `application/json`. Some HTTP libraries has convenient methods for doing that.

Reply:
Yes, that's why I check encoding type to be prepared for case when influx would stop sending us JSON (and start something else)
```
final Charset encoding =
        encodingHeader == null
            ? StandardCharsets.UTF_8
            : Charsets.toCharset(encodingHeader.getValue());
```

Reply:
Isn't it possible that you get UTF-8-encoded data, but not necessarily JSON?